packages/compiler/src/md.ts

```diff
@@ -35,10 +45,12 @@ ${content}
 
 export interface TargetMarkdownOptions {
     extraNewlineCount?: number;
+    useYaml?: boolean;
 }
 
 const defaultOptions: TargetMarkdownOptions = {
     extraNewlineCount: 1,
+    useYaml: false,
```

Comment:
There should be more granularity to `useYaml`: for example, we generally want json for vega plugins. so instead of a prop named useYaml: boolean lets have a prop named pluginFormat that takes a map of plugin names to "json" | "yaml". so we can add more serializers in the future. there should be a wildcard. and we will have a default that should be mixed in. the default is {"*": "yaml", "vega": "json", "vega-lite": "json"}